Release 5.3

fredericDelaporte opened the pull request:
It is about time to prepare this release I think.

Of course we may merge some more PR, but I think only a few more.

bahusoid commented:
Is it last 5.x release? Seems time for 6.0.

How about including multitenancy support in 5.3 (https://github.com/nhibernate/nhibernate-core/pull/2108)?

fredericDelaporte commented:
Of course having multi-tenancy would be great, but how far is it from being done?

I consider the 5.3 has waited quite too long already. It is one year and a half after 5.2 initial release.
Many other things have been contributed into 5.3, and having to wait long for getting them released is not good for contributors engagement.

I am not personally settled for going to 6.0 after 5.3.
We could have a 5.4, with a shorter development time hopefully, like less than 6 months or even only 3 months. Why not after all?

bahusoid commented:
> Of course having multi-tenancy would be great, but how far is it from being done?

Well I consider it's ready :)

lofcz commented:
I'm just a user of NHibernate but I'd like to voice support of including multitenancy in 5.3, it's a long awaited feature by many users and I'd personally rather wait a week / two than three months best case scenario for it.

bahusoid commented on releasenotes.txt:
```diff
@@ -1,4 +1,257 @@
+   * #2286 Wrong sql if used joined-subclass with filters for key columns
+   * #2278 IInterceptor.OnPrepareStatement results not used in insert/update commands
+   * #2266 Fix comment for Restrictions.IsEmpty
+   * #2255 Fix a flaky test
```
Is it manually grouping by tasks? Or somehow automatic (like Labels based)? This one seems should go to Tests. Also I see few other test related tasks in Improvements (by search test) - I think they should also go to Tests.

hazzik replied:
@bahusoid automatic

fredericDelaporte replied:
And based on `t:` labels.

bahusoid replied:
So should we change `t:` labels for test related tasks like this one? As it's not very useful fix for end user.

fredericDelaporte replied:
It is automatic for first generation, after that I manually update it, because the tool adds more than what we need and there is some cleanup to do.

Anyway if editing the release notes for changing the category of an issue, it is better to keep the issue typed accordingly.

I do not really mind where these test related tasks are categorized, provided it is not wrong.

fairking commented:
For me OData full support which was done **thanks to all of you guys**. That's what I am waiting for. 👍

fredericDelaporte commented:
@fairking, about `ConfigurationManager`, is #2116 enough for your friend case to be fixed?

fredericDelaporte commented:
We are one month later, and well, our overall contributing activity seems to me decreasing.

The milestone has two PR which are waiting since long: #2365 and #2010. They both have conflicts with current master, @maca88. #2010 is validated by me, but has a pending review request asked to @bahusoid. #2365 was validated by me but has had some additional comments by @hazzik having incurred changes, without Alexander having acknowledge these changes for now.

Bahusoid is also eager to have #2108 into 5.3, but Gunnar and me have expressed some concerns about how it works, and on my side I remain unconvinced. So I think other contributors should have a look and share their own thoughts.

Should we go on waiting on these PR?

hazzik commented:
@fredericDelaporte #2365 and #2010 are fine for me. I don't have an opinion on #2108 (I never understood uber-monolithic applications where tenants could be switched at runtime) so would rely on others to approve.

ToGoOrNotToGo commented:
As a longtime user of NHibernate, I would like to vote to complete release 5.3 now. You guys have done a great job and fixed important bugs (e.g. [NH-3865](https://github.com/nhibernate/nhibernate-core/issues/1047) in 2018) and features. Thanks a lot!

fredericDelaporte commented:
Rebased and squashed.
A few documentation PR remains in the milestone, but I think we can release without them: we could still merge them later and republish the documentation, without having to do a new release.

I intend to proceed with the release on next Sunday, if this PR is approved.
